### PR TITLE
Increase test coverage for snowflake_grants.py

### PR DESCRIPTION
**Pull request by blake-enyart**
Added additional coverage for:
* Database GRANTS/REVOKES
* Schema GRANTS/REVOKES
* Table/View GRANTS/REVOKES

### CODE REVIEW

File: src/permifrost/core/permissions/utils/snowflake_grants.py [hunk 1]
```diff
@@ -574,7 +574,7 @@ def generate_database_grants(
         for granted_database in self.grants_to_role.get(role, {}).get(
             "monitor", {}
         ).get("database", []) + self.grants_to_role.get(role, {}).get(
-            "create_schema", {}
+            "create schema", {}
```
Comment (PedramNavid) [inline src/permifrost/core/permissions/utils/snowflake_grants.py:577]: I really hate all these gets. (just venting, not directed at you)

File: src/permifrost/core/permissions/utils/snowflake_grants.py [hunk 2]
```diff
@@ -947,10 +947,13 @@ def _generate_table_read_grants(self, conn, tables, shared_dbs, role):
             # Split the table identifier into parts {DB_NAME}.{SCHEMA_NAME}.{TABLE_NAME}
             # so that we can check and use each one
             name_parts = table.split(".")
+            database_name = name_parts[0] if 0 < len(name_parts) else None
+            schema_name = name_parts[1] if 1 < len(name_parts) else None
+            table_view_name = name_parts[2] if 2 < len(name_parts) else None
```
Comment (PedramNavid) [inline src/permifrost/core/permissions/utils/snowflake_grants.py:952]: this is cute

**Review by PedramNavid — APPROVED** [review]
Looks good!

File: src/permifrost/core/permissions/utils/snowflake_grants.py (hunk 2 above)
Comment (blake-enyart) [reply src/permifrost/core/permissions/utils/snowflake_grants.py:952]: I thought so for trying to optionally get names from the name_parts section as I wasn't sure if each list was the same length 😄

File: src/permifrost/core/permissions/utils/snowflake_grants.py (hunk 1 above)
Comment (blake-enyart) [reply src/permifrost/core/permissions/utils/snowflake_grants.py:577]: You and me both!

**Pull request merged**